core/src/retries.rs

```diff
@@ -485,4 +485,68 @@ mod tests {
         listener_task.abort();
         assert!(listener_task.await.unwrap_err().is_cancelled());
     }
+
+    #[tokio::test]
+    async fn http_retry_server_json_problem() {
+        install_test_trace_subscriber();
+        let mut server = mockito::Server::new_async().await;
+
+        let mock_problem = server
+            .mock("GET", "/")
+            .with_status(400)
+            .with_header("Content-Type", "application/problem+json")
```

Comment:
nit: we have `static PROBLEM_DETAILS_JSON_MEDIA_TYPE: &str = "application/problem+json";` in `problem_details.rs` but I suppose it's not visible from here. Plus we reproduce this string literal all over our tests already, so this is no big deal.

Reply:
Updated to point to the `http_api_problem` crate's static.